### PR TITLE
cli: error management

### PR DESCRIPTION
**Pull request by heafalan**

### CODE REVIEW

File: packages/cli/main.js [hunk 1]
```diff
@@ -15,6 +15,17 @@ async function main() {
       .readAll("utf-8")
   );
 
-  console.log(response.result);
+  if (response.result !== undefined) {
```
Comment (julien-f) [inline packages/cli/main.js]: Test `response.type` instead.

File: packages/cli/main.js [hunk 2]
```diff
@@ -15,6 +15,17 @@ async function main() {
...
+    console.log(response.result);
+  }
+
+  if (response.error !== undefined) {
+    if (response.error.code === -32601) {
```
Comment (julien-f) [inline packages/cli/main.js]: Don't handle errors specifically.

File: packages/cli/main.js [hunk 3]
```diff
@@ -15,6 +15,17 @@ async function main() {
       .readAll("utf-8")
   );
 
-  console.log(response.result);
+  if (response.result !== undefined) {
+    console.log(response.result);
+  }
+
+  if (response.error !== undefined) {
```
Comment (julien-f) [inline packages/cli/main.js]: `else if`?

File: packages/cli/main.js [hunk 4]
```diff
@@ -15,6 +15,12 @@ async function main() {
       .readAll("utf-8")
   );
 
-  console.log(response.result);
+  if (response.type === "response") {
+    if (response.result !== undefined) {
```
Comment (julien-f) [inline packages/cli/main.js]: Why this test?

File: packages/cli/main.js [hunk 5]
```diff
@@ -15,6 +15,12 @@ async function main() {
...
+      console.log(response.result);
+    }
+  } else if (response.type === "error") {
+    console.log(response.error.message + " - " + response.error.data);
```
Comment (julien-f) [inline packages/cli/main.js]: Use `console.error` to display errors.

Comment (julien-f) [inline packages/cli/main.js]: `data` is not guaranteed to be a string, thus you should avoid concatenating.

```js
console.error(response.error)
```

File: packages/cli/main.js (hunk 4 above)
Comment (heafalan) [reply packages/cli/main.js]: Because the return of "deleteEntry" or "updateEntry" is undefined. It's useless to post it.

Comment (julien-f) [reply packages/cli/main.js]: I see, unfortunately this is not valid according to the JSON-RPC 2 specification, there should always be result, be it `null` or `true`, even though `json-rpc-protocol` does not enforce this for the moment.

Please fix the server accordingly.

File: packages/server/main.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ const METHODS = {
     if (!entries.delete(+id)) {
       throw new Error(`could not find entry ${id}`);
     }
+    return null;
```
Comment (julien-f) [inline packages/server/main.js]: IMHO, you should handle it outside the API method.

**Review by julien-f — APPROVED** [review]

**Pull request merged**